protocol/x/clob/keeper/orders.go

```diff
@@ -46,6 +46,83 @@ func (k Keeper) GetOperations(ctx sdk.Context) *types.MsgProposedOperations {
 	return msgProposedOperations
 }
 
+// BatchCancelShortTermOrder removes a specified batch of short term orders from all order-related data
+// structures in the memclob. As well, BatchCancelShortTermOrder adds (or updates) cancels to the desired
+// `goodTilBlock` in the memclob for all specified orders.
+// This message is not atomic. It will optimistically call `CancelShortTermOrder` for every order in the batch.
+// If any of the orders error, the error will be silently logged. This msg will only error if:
+// - Stateful validation fails
+// - Every cancel in the batch fails.
+// This method assumes the provided MsgBatchCancel has already passed ValidateBasic in CheckTx.
+func (k Keeper) BatchCancelShortTermOrder(
+	ctx sdk.Context,
+	msg *types.MsgBatchCancel,
+) error {
+	lib.AssertCheckTxMode(ctx)
+	// Note that we add `+1` here to account for the fact that `ctx.BlockHeight()` is technically the
+	// previously mined block, not the next block that will be proposed. This is due to the fact that
+	// this function is only ever called during `CheckTx`.
+	nextBlockHeight := lib.MustConvertIntegerToUint32(ctx.BlockHeight() + 1)
+
+	// Statefully validate the GTB and the clob pair ids.
+	goodTilBlock := msg.GetGoodTilBlock()
+	if err := k.validateGoodTilBlock(goodTilBlock, nextBlockHeight); err != nil {
+		return err
+	}
+	for _, batchOrder := range msg.GetShortTermCancels() {
+		clobPairId := batchOrder.GetClobPairId()
+		if _, found := k.GetClobPair(ctx, types.ClobPairId(clobPairId)); !found {
+			return errorsmod.Wrapf(
+				types.ErrInvalidClobPairParameter,
+				"Invalid clob pair id %+v",
+				clobPairId,
+			)
+		}
+	}
+	subaccountId := msg.GetSubaccountId()
+	oneCancelSucceeded := false
+	for _, batchOrder := range msg.GetShortTermCancels() {
+		clobPairId := batchOrder.GetClobPairId()
+		for _, clientId := range batchOrder.GetClientIds() {
+			msgCancelOrder := types.MsgCancelOrder{
+				OrderId: types.OrderId{
+					SubaccountId: subaccountId,
+					OrderFlags:   types.OrderIdFlags_ShortTerm,
+					ClobPairId:   clobPairId,
+					ClientId:     clientId,
+				},
+				GoodTilOneof: &types.MsgCancelOrder_GoodTilBlock{
+					GoodTilBlock: goodTilBlock,
+				},
+			}
+			// Run the short term order. If it errors, just log silently.
```

Comment:
instead of logging silently we also have the option to wrap and return it back to the user 

something like 

```
var wrapped error
for {
    ...
    if err != nil {
        wrapped = errorsmod.Wrapf(wrapped, error, ...)
    }
}
```

Reply:
ah I guess we made the decision to only return error when all cancellations fail?